arangod/V8Server/v8-collection.cpp

```diff
@@ -1880,18 +1880,43 @@ static void InsertVocbaseCol(v8::Isolate* isolate,
     }
   }
 
+
   OperationOptions options;
   if (argLength > optsIdx && args[optsIdx]->IsObject()) {
     v8::Handle<v8::Object> optionsObject = args[optsIdx].As<v8::Object>();
+
     TRI_GET_GLOBAL_STRING(WaitForSyncKey);
     if (TRI_HasProperty(context, isolate, optionsObject, WaitForSyncKey)) {
       options.waitForSync =
           TRI_ObjectToBoolean(isolate, optionsObject->Get(WaitForSyncKey));
     }
+
     TRI_GET_GLOBAL_STRING(OverwriteKey);
-    if (TRI_HasProperty(context, isolate, optionsObject, OverwriteKey)) {
+    if (!options.overwrite && TRI_HasProperty(context, isolate, optionsObject, OverwriteKey)) {
```

Comment:
IMHO `options.overwrite` is always false here, so it doesn't need to be checked.
```suggestion
    if (TRI_HasProperty(context, isolate, optionsObject, OverwriteKey)) {
```

Reply:
It could be set implicitly by overwriteMode. Therefor I want the "update" / "replace" to work even if overwrite accidentally set to false.